txscript/sighash.go

```diff
@@ -289,7 +292,14 @@ func calcWitnessSignatureHashRaw(subScript []byte, sigHashes *TxSigHashes,
 	binary.LittleEndian.PutUint32(bHashType[:], uint32(hashType))
 	sigHash.Write(bHashType[:])
 
-	return chainhash.DoubleHashB(sigHash.Bytes()), nil
+	sigHashBytes := chainhash.DoubleHashRaw(func(w io.Writer) error {
+		// TODO(rosabeef): put entire calc func into this? then no
```

Comment:
I think we should do this, yes. Also, there's another instance of `DoubleHashB` just a few lines above.

Reply:
Slight decrease here in time/op: 
```
⛰   benchstat old.txt new.txt
name                  old time/op    new time/op    delta
CalcWitnessSigHash-8    31.9µs ± 2%    31.5µs ± 1%  -1.31%  (p=0.000 n=10+10)

name                  old alloc/op   new alloc/op   delta
CalcWitnessSigHash-8    19.9kB ± 0%    19.9kB ± 0%    ~     (all equal)

name                  old allocs/op  new allocs/op  delta
CalcWitnessSigHash-8       801 ± 0%       801 ± 0%    ~     (all equal)
```